### ISSUE TITLE
QT engine has no support for css flex property?

### ISSUE THREAD

**Issue by khan789** [wkhtmltopdf/wkhtmltopdf#2866]
Hi, I simply tried to print the below html which has flex styles in it. But wkhtmltopdf library ignoring the styles. Anyone know if QT engine supports flex property? if not any alternative to achieve it?

``` javascript
<!DOCTYPE html>
<html>
<head>
<style> 
#main {
    width: 200px;
    height: 200px;
    border: 1px solid #c3c3c3;
    display: flex;
 }

#main div {
    width: 50px;
    height: 50px;
}
</style>
</head>
<body>

<div id="main">
  <div style="background-color:coral;">A</div>
  <div style="background-color:lightblue;">B</div>
  <div style="background-color:khaki;">C</div>
  <div style="background-color:pink;">D</div>
  <div style="background-color:lightgrey;">E</div>
  <div style="background-color:lightgreen;">F</div>
</div>

<p><b>Note:</b> Internet Explorer 10 and earlier versions do not support the flex-wrap property.</p>

<p><b>Note:</b> Safari 6.1 (and newer) supports an alternative, the -webkit-flex-wrap property.</p>

</body>
</html>
```

**Comment by khan789** [comment]
Sorry, i found the answer in the other thread. obviously flex is not supported in the current version of the QT engine.  https://github.com/wkhtmltopdf/wkhtmltopdf/issues/1522 

I'm closing the issue now.

**Comment by mkhazov** [comment]
@khan789 you accidentally reopened it

**Comment by andyyou** [comment]
After I upgrade my wkhtmltopdf to `0.12.5 (with patched qt)` in `Ubuntu 16.04` and `OSX 10.13.5（17F77)`. I found out `Flex` is not working.

This is my sample html

```
<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <style>
    .row {
      display: flex;
      flex-direction: row;
      justify-content: center;
    }
    .col-1 {
      flex: 1;
      background-color: red;
    }
    .col-2 {
      flex: 2;
      background-color: blue;
    }
    </style>
  </head>

  <body>
    <div class="row">
      <div class="col-1">1</div>
      <div class="col-2">2</div>
    </div>
  </body>
</html>
```

![2018-07-08 20-28-29](https://user-images.githubusercontent.com/665690/42419755-8c696900-82ed-11e8-84fd-5ab5c074f908.png)

Is that same problem?
